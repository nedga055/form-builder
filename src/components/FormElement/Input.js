import React, { Component } from 'react';

import PropTypes from 'prop-types';

class Input extends Component {
	constructor(props) {
		super(props);

		this.state = {
			focused: false
		};

		this.onFocus = this.onFocus.bind(this);
		this.onBlur = this.onBlur.bind(this);
	}

	onFocus() {
		this.setState({
			focused: true
		});
	}

	onBlur() {
		this.setState({
			focused: false
		});
	}

	render() {
		const { id, name, type, placeholder, label, required, disabled } = this.props;

		const { focused } = this.state;

		let fgLineClasses = (!focused) ? 'fg-line' : 'fg-line fg-toggled';

		return (
			<div className="form-group">
				<div className={fgLineClasses}>
					<label className="sr-only" htmlFor={name}>{label}</label>
					<input id={id}
					       className="form-control"
					       name={name}
					       type={type}
					       placeholder={placeholder}
					       required={required}
					       disabled={disabled}
					       onFocus={this.onFocus}
					       onBlur={this.onBlur}
					/>
				</div>
			</div>
		)
	}
}

Input.defaultProps = {
	id: '',
	name: '',
	type: 'text',
	placeholder: '',
	label: '',
	required: true,
	disabled: false
};

Input.propTypes = {
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	type: PropTypes.string,
	placeholder: PropTypes.string,
	label: PropTypes.string.isRequired,
	required: PropTypes.bool,
	disabled: PropTypes.bool
};

export default Input;