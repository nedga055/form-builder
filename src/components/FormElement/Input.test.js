import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('<Input/>', () => {
	it('input rendered correctly', () => {
		const component = renderer.create(
			<Input id="add-component" name="add-component" placeholder="Add a new component" label="Add a new component" />
		);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('focus changes wrapper class', () => {
		const wrapper = shallow(<Input id="add-component" name="add-component" placeholder="Add a new component" label="Add a new component" />);

		expect(wrapper.find('.fg-line').hasClass('fg-toggled')).toEqual(false);

		wrapper.find('input').simulate('focus');
		expect(wrapper.find('.fg-line').hasClass('fg-toggled')).toEqual(true);

		wrapper.find('input').simulate('blur');
		expect(wrapper.find('.fg-line').hasClass('fg-toggled')).toEqual(false);
	});
});