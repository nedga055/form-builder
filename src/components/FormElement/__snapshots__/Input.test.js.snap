// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> input rendered correctly 1`] = `
<div
  className="form-group"
>
  <div
    className="fg-line"
  >
    <label
      className="sr-only"
      htmlFor="add-component"
    >
      Add a new component
    </label>
    <input
      className="form-control"
      disabled={false}
      id="add-component"
      name="add-component"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Add a new component"
      required={true}
      type="text"
    />
  </div>
</div>
`;
